front/lib/api/llm/clients/google/types.ts

```diff
@@ -1,52 +1,90 @@
+import flatMap from "lodash/flatMap";
+
+import type { LLMParameters } from "@app/lib/api/llm/types/options";
 import type { ModelIdType } from "@app/types";
 import {
   GEMINI_2_5_FLASH_LITE_MODEL_ID,
   GEMINI_2_5_FLASH_MODEL_ID,
   GEMINI_2_5_PRO_MODEL_ID,
+  GEMINI_3_PRO_MODEL_ID,
 } from "@app/types";
 
-export const GOOGLE_AI_STUDIO_WHITELISTED_NON_REASONING_MODEL_IDS: ModelIdType[] =
-  [];
+const GOOGLE_AI_STUDIO_PROVIDER_ID = "google_ai_studio";
+export const GOOGLE_AI_STUDIO_MODEL_FAMILIES = [
+  "gemini-2",
+  "gemini-3",
+] as const;
+export type GoogleAIStudioModelFamily =
+  (typeof GOOGLE_AI_STUDIO_MODEL_FAMILIES)[number];
 
-export const GOOGLE_AI_STUDIO_WHITELISTED_REASONING_MODEL_IDS: ModelIdType[] = [
-  GEMINI_2_5_FLASH_MODEL_ID,
-  GEMINI_2_5_FLASH_LITE_MODEL_ID,
-  GEMINI_2_5_PRO_MODEL_ID,
-];
-
-export const GOOGLE_AI_STUDIO_WHITELISTED_MODEL_IDS: ModelIdType[] = [
-  ...GOOGLE_AI_STUDIO_WHITELISTED_NON_REASONING_MODEL_IDS,
-  ...GOOGLE_AI_STUDIO_WHITELISTED_REASONING_MODEL_IDS,
-];
+export const GOOGLE_AI_STUDIO_MODEL_FAMILY_CONFIGS = {
+  "gemini-2": {
+    modelIds: [
+      GEMINI_2_5_FLASH_MODEL_ID,
+      GEMINI_2_5_FLASH_LITE_MODEL_ID,
+      GEMINI_2_5_PRO_MODEL_ID,
+    ],
+    overwrites: {},
+  },
+  "gemini-3": {
+    modelIds: [GEMINI_3_PRO_MODEL_ID],
+    overwrites: {
+      // Not required but strongly recommended by Google for Gemini 3
```

Comment:
Interesting that it's not required. Probably the only reasoning model that still supports it !